src/vcpkg/commands.build.cpp

```diff
@@ -1690,30 +1690,110 @@ namespace vcpkg
         return "https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+" + spec_name;
     }
 
-    static std::string make_gh_issue_open_url(StringView spec_name, StringView triplet, StringView path)
+    static std::string make_gh_issue_open_url(StringView spec_name, StringView triplet, StringView body)
     {
         return Strings::concat("https://github.com/microsoft/vcpkg/issues/new?title=[",
                                spec_name,
                                "]+Build+error+on+",
                                triplet,
-                               "&body=Copy+issue+body+from+",
-                               Strings::percent_encode(path));
+                               "&body=",
+                               Strings::percent_encode(body));
+    }
+
+    enum class CIType
+    {
+        GithubActions,
+        GitLabCi,
+        Azure,
+    };
+
+    static Optional<CIType> detect_ci_type()
+    {
+        if (get_environment_variable(EnvironmentVariableGitHubActions).has_value())
+        {
+            return CIType::GithubActions;
+        }
+        if (get_environment_variable(EnvironmentVariableGitLabCI).has_value())
+        {
+            return CIType::GitLabCi;
+        }
+        if (get_environment_variable(EnvironmentVariableTfBuild).has_value())
+        {
+            return CIType::Azure;
+        }
+        return {};
+    }
+
+    static void append_file_collapsible(LocalizedString& output,
+                                        CIType type,
+                                        const ReadOnlyFilesystem& fs,
+                                        const Path& file)
+    {
+        auto title = file.filename();
+        // starting tag
+        if (type == CIType::GithubActions)
+        {
+            // https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#grouping-log-lines
+            output.append_raw("::group::").append_raw(title).append_raw('\n');
+        }
+        else if (type == CIType::GitLabCi)
+        {
+            // https://docs.gitlab.com/ee/ci/jobs/job_logs.html#custom-collapsible-sections
+            using namespace std::chrono;
+            const auto timestamp = duration_cast<seconds>(system_clock::now().time_since_epoch()).count();
+            output.append_raw(fmt::format("\\e[0Ksection_start:{}:SECTION_NAME[collapsed=true]\r\\e[0K", timestamp))
```

Comment:
`SECTION_NAME` is almost certainly incorrect; the example docs in the URI replaced that with a real name. Do you have an example of a test instance that has used this?